lib/src/tree_controller.dart

```diff
@@ -296,6 +254,44 @@ class TreeController<T extends Object> with ChangeNotifier {
     rebuild();
   }
 
+  /// Whether all root nodes of this tree are expanded.
+  bool get areAllRootsExpanded => roots.every(getExpansionState);
+
+  /// Whether all root nodes of this tree are collapsed.
+  bool get areAllRootsCollapsed => !roots.any(getExpansionState);
+
+  /// Whether **all** nodes of this tree are expanded.
+  bool get isTreeExpanded {
+    if (roots.isEmpty) return false;
+
+    bool anyCollapsedNodes = true;
+
+    breadthFirstSearch(
+      returnCondition: (T node) {
+        anyCollapsedNodes = !getExpansionState(node);
+        return anyCollapsedNodes;
+      },
+    );
+
+    return anyCollapsedNodes;
+  }
+
+  /// Whether **all** nodes of this tree are collapsed.
+  bool get isTreeCollapsed {
+    if (roots.isEmpty) return true;
```

Comment:
Hi, @baumths. I am curious if this condition should return `true`? I think if there are no roots then tree is in "undefined" state, i.e. it is neither expanded or collapsed.

Reply:
I'm actually not sure what the behavior should be, if there is no nodes on the tree it could be considered collapsed? Or I could remove the check all together and expect the user to do this check?

Btw, I haven't had time to test this code, not sure it works yet... I'll work on this PR later today.